### PR TITLE
Tests: implement tests for QCoroImageProvider

### PR DESCRIPTION
**Pull request by danvratil**

### CODE REVIEW

**Comment by JBBgameich** [comment]
Maybe `export QT_QPA_PLATFORM=offscreen` could work

File: tests/qcoroimageprovider.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
+// SPDX-FileCopyrightText: 2021 Daniel Vrátil <dvratil@kde.org>
```
Comment (JBBgameich) [inline tests/qcoroimageprovider.cpp:1]:
```suggestion
// SPDX-FileCopyrightText: 2022 Daniel Vrátil <dvratil@kde.org>
```

**Comment by danvratil** [comment]
Thanks for the tip with offscreen QPA, @JBBgameich!

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -20,7 +20,11 @@ function(qcoro_add_test _name)
     )
     if (NOT TEST_SKIP_ADD_TEST)
         add_test(NAME test-${_name} COMMAND test-${_name})
+        set_tests_properties(test-${_name} PROPERTIES
+            ENVIRONMENT LSAN_OPTIONS=suppressions=${CMAKE_CURRENT_SOURCE_DIR}/${_name}-qt$QT_VERSION_MAJOR}-lsan.supp
```
Comment (JBBgameich) [inline tests/CMakeLists.txt:24]:
```suggestion
            ENVIRONMENT LSAN_OPTIONS=suppressions=${CMAKE_CURRENT_SOURCE_DIR}/${_name}-qt${QT_VERSION_MAJOR}-lsan.supp
```

**Pull request merged**